### PR TITLE
fix: unit and integration tests

### PR DESCRIPTION
**Pull request by DanielSchiavini**
image

### CODE REVIEW

**Review by charles-cooper — APPROVED** [review]
looks good. can we swap the order of BoaLoader and BoaImporter so it follows the logic / execution flow?

File: boa/interpret.py [hunk 1]
```diff
@@ -26,39 +28,30 @@
...
-    def load_module(self, fullname):
-        if fullname in sys.modules:
-            return sys.modules[fullname]
 
-        # import system should guarantee this, but be paranoid
-        if fullname not in self._path_lookup:
-            raise ImportError(f"invariant violated: no lookup for {fullname}")
+class BoaLoader(SourceFileLoader):
+    def get_code(self, fullname):
+        return ""
```
Comment (charles-cooper) [inline boa/interpret.py:45]: per https://docs.python.org/3/library/importlib.html#importlib.abc.InspectLoader.get_code, this should return None

Comment (DanielSchiavini) [reply boa/interpret.py:45]: image

Comment (charles-cooper) [reply boa/interpret.py:45]: nvm, discussed offline -- apparently returning `None` triggers an `ImportError`.

Comment (charles-cooper) [reply boa/interpret.py:45]: let's add a comment though
```suggestion
        # importlib docs say to return None, but that triggers an `ImportError`
        return ""
```

**Pull request merged**